tests/test_workspaces.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use itertools::Itertools;
+
+use crate::common::TestEnvironment;
+
+pub mod common;
+
+/// Test adding a second workspace
+#[test]
+fn test_workspaces_add_second_workspace() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "--git", "main"]);
```

Comment:
I assume this is like ASSERT_FOO in that it somehow aborts out of the test if it's not successful?

Reply:
There are no macros involved here (which are indicated with a trailing `!`), so this isn't expanding to some `return`-early construct. Most likely it's `panic`king and the panic is caught by the test runner (which will proceed to run other tests; it won't abort the entire test running process).